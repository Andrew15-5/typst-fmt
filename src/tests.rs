use super::*;

fn init() {
    let _ = env_logger::builder()
        .filter_level(log::LevelFilter::Debug)
        .is_test(true)
        .try_init();
}

macro_rules! make_test {
    ($test_name:ident, $input:literal) => {
        make_test!($test_name, $input, Config::default());
    };

    ($test_name:ident, $input:literal, $config:expr) => {
        #[test]
        fn $test_name() {
            init();
            let input = $input;
            let formatted = format(input, $config);
            insta::with_settings!({description => format!("INPUT\n===\n{input:?}\n===\n{input}\n===\nFORMATTED\n===\n{formatted}")}, {
                insta::assert_debug_snapshot!(formatted);
            });
        }
    };

    }

make_test!(unchanged, "#{ }");
make_test!(on_space, "#{  }");
make_test!(on_space_b, "#{   }");
make_test!(two_line_max, "\n\n\n");
make_test!(call_func_empty, "#f()");
make_test!(call_func_simple, "#f(1,2,3)");
make_test!(
    call_func_long,
    "#f(1,this_is_absurdly_loooooooooong,3)",
    Config {
        max_line_length: 1,
        ..Default::default()
    }
);
make_test!(
    call_func_long_trailing,
    "#f(1,this_is_absurdly_loooooooooong,3,)",
    Config {
        max_line_length: 1,
        ..Default::default()
    }
);
