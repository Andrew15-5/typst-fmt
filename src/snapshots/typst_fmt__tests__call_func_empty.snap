---
source: src/tests.rs
description: "INPUT\n===\n\"#f()\"\n===\n#f()\nFORMATTED\n===\n#f()"
expression: formatted
---
"#f()"
